src/dataservice/data.go

```diff
@@ -0,0 +1,330 @@
+
+package main
+
+import (
+        "bytes"
+        "encoding/json"
+        "flag"
+        "fmt"
+        "io/ioutil"
+        "math/rand"
+        "net/http"
+        "net/url"
+        "strconv"
+        "time"
+
+        "github.com/vmware/harbor/src/common/models"
+        "github.com/manveru/faker"
+)
+
+// Harbor deployment URL
+var baseURL string
+
+// This instantiates a Faker module used to generate randomized data
+func getInstance() *faker.Faker {
+
+    fake, err := faker.New("en")
+    if err != nil {
+      panic(err)
+    }
+    return fake
+}
+
+func generateUser(role string) models.User {
+    faker := getInstance()
+    var name = faker.Name()
+    for len(name) > 20 {
+        name = faker.Name()
+    }
+    m := models.User{
+            Username:       faker.UserName(),
+            Email:          faker.Email(),
+            Password:       "Harbor12345",
+            Realname:       name,
+            Comment:        faker.Characters(20),
+    }
+    return m
+}
+
+// This method logs in and returns the Session ID
+func login(baseURL, username, password string) string {
+    resource := "/login"
+    data := url.Values{}
+    data.Set("principal", username)
+    data.Add("password", password)
+
+    u, _ := url.ParseRequestURI(baseURL)
+    u.Path = resource
+    urlStr := fmt.Sprintf("%v", u)
+    client := &http.Client{}
+    r, _ := http.NewRequest("POST", urlStr, bytes.NewBufferString(data.Encode()))
+    r.Header.Add("Content-Type", "application/x-www-form-urlencoded")
+    r.Header.Add("Content-Length", strconv.Itoa(len(data.Encode())))
+
+    resp, _ := client.Do(r)
+    return resp.Cookies()[0].Value
+}
+
+func getJSON(baseURL string, endpoint string, cookie http.Cookie) *http.Response {
+
```

Comment:
nit: There are several places with the leading blank line in functions, can we remove them?